File: Source/ContactManager/Models/Domain/RequestModel.vb

Imports Contensive.BaseClasses
''' <summary>
''' requests processed by this form
''' </summary>
Public Class RequestModel
    Public Property TabNumber As Integer
    Public Property Button As String
    Public Property DetailMemberID As Integer
    Public Property FormID As FormIdEnum
    Public Property RowCount As Integer
    Public Property GroupID As Integer
    Public Property GroupToolAction As GroupToolActionEnum
    Public Property GroupToolSelect As Integer
    Public Property SelectionGroupSubTab As String
    Public Property SelectionSearchSubTab As String
    '
    Public Sub New(cp As CPBaseClass)
        '
        ' todo - convert request names to constants
        '
        TabNumber = cp.Doc.GetInteger("tab")
        Button = cp.Doc.GetText("Button")
        DetailMemberID = cp.Doc.GetInteger(RequestNameMemberID)
        RowCount = cp.Doc.GetInteger("M.Count")
        GroupID = cp.Doc.GetInteger("GroupID")
        GroupToolSelect = cp.Doc.GetInteger("GroupToolSelect")
        SelectionGroupSubTab = cp.Doc.GetText("SelectionGroupSubTab")
        SelectionSearchSubTab = cp.Doc.GetText("SelectionSearchSubTab")
        '
        Dim testAction As Integer = cp.Doc.GetInteger("GroupToolAction")
        GroupToolAction = If((testAction > 0 And testAction < 5), CType(testAction, GroupToolActionEnum), GroupToolActionEnum.nop)
        '
        Dim testFormId As Integer = cp.Doc.GetInteger(RequestNameFormID)
        FormID = If((testFormId > 0 And testFormId < 4), CType(testFormId, FormIdEnum), FormIdEnum.FormUnknown)

    End Sub
End Class

File: Source/ContactManager/Models/Domain/RequestModel.cs

using Contensive.BaseClasses;

namespace Contensive.Addons.ContactManager {
    /// <summary>
/// requests processed by this form
/// </summary>
    public class RequestModel {
        public int TabNumber { get; set; }
        public string Button { get; set; }
        public int DetailMemberID { get; set; }
        public Constants.FormIdEnum FormID { get; set; }
        public int RowCount { get; set; }
        public int GroupID { get; set; }
        public Constants.GroupToolActionEnum GroupToolAction { get; set; }
        public int GroupToolSelect { get; set; }
        public string SelectionGroupSubTab { get; set; }
        public string SelectionSearchSubTab { get; set; }
        // 
        public RequestModel(CPBaseClass cp) {
            // 
            // todo - convert request names to Constants
            // 
            TabNumber = cp.Doc.GetInteger("tab");
            Button = cp.Doc.GetText("Button");
            DetailMemberID = cp.Doc.GetInteger(Constants.RequestNameMemberID);
            RowCount = cp.Doc.GetInteger("M.Count");
            GroupID = cp.Doc.GetInteger("GroupID");
            GroupToolSelect = cp.Doc.GetInteger("GroupToolSelect");
            SelectionGroupSubTab = cp.Doc.GetText("SelectionGroupSubTab");
            SelectionSearchSubTab = cp.Doc.GetText("SelectionSearchSubTab");
            // 
            int testAction = cp.Doc.GetInteger("GroupToolAction");
            GroupToolAction = testAction > 0 & testAction < 5 ? (Constants.GroupToolActionEnum)testAction : Constants.GroupToolActionEnum.nop;
            // 
            int testFormId = cp.Doc.GetInteger(Constants.RequestNameFormID);
            FormID = testFormId > 0 & testFormId < 4 ? (Constants.FormIdEnum)testFormId : Constants.FormIdEnum.FormUnknown;

        }
    }
}